From fdd95aaadd62d10b8bcce8cc2745a42ff190a54d Mon Sep 17 00:00:00 2001
From: Caleb Crome <caleb@crome.org>
Date: Thu, 7 Jan 2016 17:41:13 -0800
Subject: [PATCH 1/3] ASoC: soc_utils: add devicetree bindings to dummy driver

Allow device tree to specify a dummy driver with no hardware, but use the SSI port

Arnaud's patch

Signed-off-by: Caleb Crome <caleb@crome.org>
---
 sound/soc/soc-utils.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/sound/soc/soc-utils.c b/sound/soc/soc-utils.c
index e30aacb..bcc9efc 100644
--- a/sound/soc/soc-utils.c
+++ b/sound/soc/soc-utils.c
@@ -19,6 +19,7 @@
 #include <sound/pcm.h>
 #include <sound/pcm_params.h>
 #include <sound/soc.h>
+#include <linux/module.h>
 
 int snd_soc_calc_frame_size(int sample_size, int channels, int tdm_slots)
 {
@@ -359,9 +360,18 @@ static int snd_soc_dummy_remove(struct platform_device *pdev)
 	return 0;
 }
 
+#ifdef CONFIG_OF
+static const struct of_device_id soc_dummy_ids[] = {
+    { .compatible = "linux,snd-soc-dummy", },
+    { }
+};
+MODULE_DEVICE_TABLE(of, soc_dummy_ids);
+#endif
+
 static struct platform_driver soc_dummy_driver = {
 	.driver = {
 		.name = "snd-soc-dummy",
+		.of_match_table = of_match_ptr(soc_dummy_ids),
 	},
 	.probe = snd_soc_dummy_probe,
 	.remove = snd_soc_dummy_remove,
-- 
2.7.4

